### PR TITLE
[batch] enable Stackdriver Profiler for batch-driver

### PR DESCRIPTION
**Pull request by danking**
# Changes
- include the GCP profiler utility in a new service-base only docker requirements
- include necessary `build-essentials` (in particular, `gcc`) in the service-base image
- enable profiler utility for batch driver when HAIL_SHOULD_PROFILE is set, which is set on dev deploy or deploy

# Profiler Information
We have [enough quota](https://console.cloud.google.com/iam-admin/quotas?_ga=2.207444499.1054557266.1580858577-1098760465.1578424762&project=hail-vdc&folder&organizationId=548622027621&service=cloudprofiler.googleapis.com) to handle a substantial number of parallel profiled batch-driver instances. I don't think we have enough to handle a deluge of PRs, so I disabled profiling of PRs.

Stackdriver Profiling is currently free, but we should keep an eye on it. When it becomes a paid product, we may reconsider these settings.

# Example

I ran a pipeline with 1000 `/bin/true` jobs and then went to the [profiler page](
https://console.cloud.google.com/profiler;timespan=10m;end=2020-02-03T20:50:00.000Z/batch-driver/CPU?project=hail-vdc).
Screen Shot 2020-02-03 at 3 55 55 PM

cc: @cseed

### CODE REVIEW

File: docker/requirements.txt [hunk 1]
```diff
@@ -40,3 +40,4 @@ urllib3==1.24.3
 uvloop==0.12.2
 Werkzeug==0.15.4
 zulip==0.6.3
+google-cloud-profiler==1.0.9
```
Comment (jigold) [inline docker/requirements.txt]: can you put this in alphabetical order.

**Comment by jigold** [comment]
Why do you need build-essential in the worker docker image? We want to keep that as slim as possible. Do you know if it's really slow/big?

**Comment by danking** [comment]
The worker image installs `docker/requirements.txt` which pip installs `googlecloudprofiler` which requires gcc.

I could probably figure out the actual requirements of the worker image and list those separately. That seems prudent anyway. However, in the long term, if we want to profiler the workers, we'd need build-essential. Installing build-essential adds 204MB.

**Comment by danking** [comment]
I'll isolate the worker's requirements from everything else tomorrow. That'll probably actually yield substantial byte savings.

**Comment by danking** [comment]
@jigold ok, the changes are isolated to service-base.

**Pull request merged**